### PR TITLE
Use AP_GPS_ENABLED to exclude more code

### PR DESCRIPTION
**Pull request by peterbarker**
Board               AP_Periph  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
Durandal                       *      *           *       *                 *      *      *
HerePro             *                                                                     
Hitec-Airspeed      *                                                                     
KakuteH7-bdshot                *      *           *       *                 *      *      *
MatekF405                      16     *           16      8                 8      16     8
Pixhawk1-1M-bdshot             *                  *       *                 *      *      *
f103-QiotekPeriph   *                                                                     
f303-Universal      *                                                                     
iomcu                                                           *                         
revo-mini                      *      *           *       *                 *      *      *
skyviper-v2450                                    *                                       
```

This work sponsored by Spektreworks

### CODE REVIEW

**Comment by amilcarlucas** [comment]
Is there a way to make the github regression tests aware of these compile flags?

It would be interresting to run the tests on configurations that have features disabled.

**Comment by peterbarker** [comment]
> Is there a way to make the github regression tests aware of these compile flags?

Yes!  `build_options.py` does that.

> It would be interresting to run the tests on configurations that have features disabled.

We do test some trivial examples of that based on `build_options.py`.

See https://github.com/ardupilot/ardupilot/blob/master/.github/workflows/test_chibios.yml#L157

**Comment by peterbarker** [comment]
> It would be interresting to run the tests on configurations that have features disabled.

.... see also `test_build_options.py` which tests a *lot* of things based on `build_options.py` - enabling each option in turn, disabling each option in turn , enabling all, enabling none and enabling defaults... it takes a while to run :-)

**Comment by amilcarlucas** [comment]
Nice, I guess I should be on the dev meetings more often. I missed that info.

File: libraries/AP_GPS/AP_GPS_config.h [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #endif
 
 #ifndef AP_GPS_DRONECAN_ENABLED
-#define AP_GPS_DRONECAN_ENABLED HAL_ENABLE_DRONECAN_DRIVERS
+#define AP_GPS_DRONECAN_ENABLED AP_GPS_BACKEND_DEFAULT_ENABLED && HAL_ENABLE_DRONECAN_DRIVERS
```
Comment (tridge) [inline libraries/AP_GPS/AP_GPS_config.h:15]: can we run the full matrix check on this? this could break a board

**Review by tridge — CHANGES_REQUESTED** [review]
just need to run the full matrix check to ensure this doesn't break a board

File: libraries/AP_GPS/AP_GPS_config.h (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_GPS/AP_GPS_config.h:15]: Done.  Of all of the boards that compile, all changes are explained (changed on-board hwdef)

**Pull request merged**